src/shmem_node_util.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include "shmem_internal.h"
+#include "shmem_node_util.h"
+
+
+int *shmem_internal_location_array = NULL;
+
+static int node_util_is_initialized = 0;
+static char my_hostname[SHMEM_INTERNAL_MAX_HOSTNAME_LEN];
+static int n_local_pes = 0;
+
+int shmem_node_util_init(void)
+{
+    int ret;
+    char errmsg[256];
+
+    if (!node_util_is_initialized) {
+        if (gethostname(my_hostname, SHMEM_INTERNAL_MAX_HOSTNAME_LEN)) {
+            RETURN_ERROR_MSG("gethostname failed '%s'", shmem_util_strerror(errno, errmsg, 256));
+            return errno;
+        }
+
+        shmem_internal_location_array = malloc(sizeof(int) * shmem_internal_num_pes);
+        if (NULL == shmem_internal_location_array) {
+            RETURN_ERROR_STR("Error: out of memory when allocating node_util location array");
```

Comment:
No need to include "Error" in the message; the macro does this for you.

Reply:
Yeah, I see a few in `transport_ofi.c` as well that I'll fix.